tritonsrc/attn_torch_function.py

```diff
@@ -471,8 +471,8 @@ def backward(ctx, do, _, fwd_tuning_result):
             BLOCK_M = 128
             BLOCK_N = 64
         if q.dtype == torch.float32:
-            BLOCK_M //= 2
-            BLOCK_N //= 2
+            BLOCK_M = max(16, BLOCK_M // 2)
```

Comment:
Is this just for FP32?  Could we gate on that dtype?

Reply:
Just for fp32. Otherwise fp32 result will fail.